test/cdn-site-hosting/cdn-site-hosting-construct.test.ts

```diff
@@ -82,4 +81,48 @@ describe("CdnSiteHostingConstruct", () => {
       );
     });
   });
+
+  describe("For a routed SPA", () => {
+    let stack: Stack;
+
+    beforeAll(() => {
+      const app = new cdk.App();
+      stack = new cdk.Stack(app, "TestRoutedSPAStack", { env: testEnv });
+      new CdnSiteHostingConstruct(stack, "MyTestConstruct", {
+        certificateArn: fakeCertificateArn,
+        siteSubDomain: fakeSiteSubDomain,
+        domainName: fakeDomain,
+        removalPolicy: RemovalPolicy.DESTROY,
+        isRoutedSpa: true,
+        sources: [s3deploy.Source.asset("./")],
+        websiteErrorDocument: "error.html",
+        websiteIndexDocument: "index.html",
+      });
+    });
+
+    test("configures a custom error response code override in CloudFront", () => {
+      expectCDK(stack).to(
+        haveResourceLike("AWS::CloudFront::Distribution", {
+          DistributionConfig: {
+            CustomErrorResponses: [
+              {
+                ErrorCode: 404,
+                ResponseCode: 200,
+                ResponsePagePath: "/index.html",
+              },
+            ],
+          },
+        })
+      );
+    });
+    test("configures an error document in S3", () => {
+      expectCDK(stack).to(
+        haveResourceLike("AWS::S3::Bucket", {
+          WebsiteConfiguration: {
+            ErrorDocument: "error.html",
+          },
+        })
+      );
+    });
+  });
```

Comment:
I've added this test here, I noticed in `cdn-site-hosting-with-dns-construct.test.ts` there's a lot of quite similar tests, should these tests be added there as well?

Reply:
I added them in both places simply because if we decouple the two at a later point, we want regression coverage.